gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/GmailUtils.java

```diff
@@ -0,0 +1,836 @@
+        if (bccRecipients != null) {
+            message.setRecipients(Message.RecipientType.BCC, InternetAddress.parse(bccRecipients));
+        }
+        message.setSubject(subject);
+        MimeMultipart content = new MimeMultipart();
+        MimeBodyPart mainPart = new MimeBodyPart();
+        mainPart.setText(textContent);
+        content.addBodyPart(mainPart);
+
+        for (String attachment : attachmentList) {
+            javax.activation.DataHandler handler = axis2MsgCtx.getAttachment(attachment);
+            if (handler != null) {
+                InputStream inStream = handler.getInputStream();
+                byte[] bytes = IOUtils.toByteArray(inStream);
+                ByteArrayDataSource source =
+                        new ByteArrayDataSource(bytes,
+                                handler.getContentType());
+                MimeBodyPart bodyPart = new MimeBodyPart();
+                bodyPart.setDataHandler(new DataHandler(source));
+                bodyPart.setFileName(attachment);
+                content.addBodyPart(bodyPart);
+            } else {
+                String errorLog = "Invalid attachemnt ID, " + attachment;
+                log.error(errorLog);
+                ConnectException connectException = new ConnectException(errorLog);
+                throw (connectException);
+            }
+        }
+        message.setContent(content);
+        return message;
+    }
+
+    /**
+     * Sends the given {@link Message} through the given {@link SMTPTransport}
+     *
+     * @param message
+     *            The message to be sent
+     * @param transport
+     *            The {@link SMTPTransport} through which the message should be
+     *            sent
+     * @throws MessagingException
+     *             as a result of failures in message transportation
+     */
+    public static void sendMessage(Message message, SMTPTransport transport)
+            throws MessagingException {
+        log.info("Sending the mail...");
+        transport.sendMessage(message, message.getAllRecipients());
+        log.info("The mail is succesfully sent");
+    }
+
+    private static Message[] getBatch(Message[] messages, int batchNumber) {
+        Message[] newMessages;
+        if (messages.length > GmailConstants.GMAIL_BATCH_SIZE * batchNumber) {
+            newMessages =
+                    Arrays.copyOfRange(messages, messages.length -
+                                    GmailConstants.GMAIL_BATCH_SIZE *
+                                            batchNumber,
+                            messages.length - GmailConstants.GMAIL_BATCH_SIZE *
+                                    (batchNumber - 1));
+        } else if (messages.length > GmailConstants.GMAIL_BATCH_SIZE * (batchNumber - 1)) {
+            newMessages =
+                    Arrays.copyOfRange(messages, 0, messages.length -
+                            GmailConstants.GMAIL_BATCH_SIZE *
+                                    (batchNumber - 1));
```

Comment:
Please formate the code properly. Above 4 lines can be in one line. This should be fixed in entire PR.